pennylane_lightning/src/algorithms/AdjointDiff.hpp

```diff
@@ -11,399 +11,145 @@
+         op_idx--) {
+        PL_ABORT_IF(ops.getOpsParams()[op_idx].size() > 1,
+                    "The operation is not supported using the adjoint "
+                    "differentiation method");
+        if ((ops_name[op_idx] == "QubitStateVector") ||
+            (ops_name[op_idx] == "BasisState")) {
+            continue; // Ignore them
         }
 
-        auto tp_it = tp.rbegin();
-        const auto tp_rend = tp.rend();
-
-        // Create observable-applied state-vectors
-        std::vector<StateVectorManagedCPU<T>> H_lambda(
-            num_observables, StateVectorManagedCPU<T>{lambda.getNumQubits()});
-        applyObservables(H_lambda, lambda, obs);
-
-        StateVectorManagedCPU<T> mu(lambda.getNumQubits());
-
-        for (int op_idx = static_cast<int>(ops_name.size() - 1); op_idx >= 0;
-             op_idx--) {
-            PL_ABORT_IF(ops.getOpsParams()[op_idx].size() > 1,
-                        "The operation is not supported using the adjoint "
-                        "differentiation method");
-            if ((ops_name[op_idx] == "QubitStateVector") ||
-                (ops_name[op_idx] == "BasisState")) {
-                continue;
-            }
-            if (tp_it == tp_rend) {
-                break; // All done
-            }
-            mu.updateData(lambda.getDataVector());
-            applyOperationAdj(lambda, ops, op_idx);
-
-            if (ops.hasParams(op_idx)) {
-                if (current_param_idx == *tp_it) {
-                    const T scalingFactor =
-                        applyGenerator(mu, ops_name[op_idx],
-                                       ops.getOpsWires()[op_idx],
-                                       !ops.getOpsInverses()[op_idx]) *
-                        (ops.getOpsInverses()[op_idx] ? -1 : 1);
-
-                    const size_t mat_row_idx =
-                        trainableParamNumber * num_observables;
-
-                    // clang-format off
-
-                    #if defined(_OPENMP)
-                        #pragma omp parallel for default(none)   \
-                        shared(H_lambda, jac, mu, scalingFactor, \
-                            mat_row_idx,        \
+        if (tp_it == tp_rend) {
+            break; // All done
+        }
+        mu.updateData(lambda.getDataVector());
+        applyOperationAdj(lambda, ops, op_idx);
+
+        if (ops.hasParams(op_idx)) {
+            if (current_param_idx == *tp_it) {
+                // if current parameter is a trainable parameter
+                const T scalingFactor =
+                    mu.applyGenerator(ops_name[op_idx],
+                                      ops.getOpsWires()[op_idx],
+                                      !ops.getOpsInverses()[op_idx]) *
+                    (ops.getOpsInverses()[op_idx] ? -1 : 1);
+
+                const size_t mat_row_idx =
+                    trainableParamNumber * num_observables;
+
+                // clang-format off
+                
```

Comment:
```suggestion
```

Reply:
Hi Amintor, thanks for the comment. I found that this line has very weird property that `clang-format` (version 12 we are using) does not correctly format the line above (the line with `// clang-format off`) without this blank. It must from a bug in `clang-format`, but I think we should keep this line at the moment (until we upgrade clang-format version).

Reply:
Oh! Weird! Thanks for the clarification!